src/modules/trade/state/tradeConfirmStateAtom.ts

```diff
@@ -1,25 +1,68 @@
 import { atom } from 'jotai'
 
-import { TradeConfirmationProps } from '../pure/TradeConfirmation'
+import { TradeAmounts } from '../types/TradeAmounts'
 
 interface TradeConfirmModalState {
-  isPending: boolean
+  isOpen: boolean
+  pendingTrade: TradeAmounts | null
   transactionHash: string | null
   error: string | null
-  confirmationState: TradeConfirmationProps | null
 }
 
 export const tradeConfirmStateAtom = atom<TradeConfirmModalState>({
-  isPending: false,
+  isOpen: false,
+  pendingTrade: null,
   transactionHash: null,
   error: null,
-  confirmationState: null,
 })
 
 export const updateTradeConfirmStateAtom = atom(null, (get, set, nextState: Partial<TradeConfirmModalState>) => {
   set(tradeConfirmStateAtom, () => {
-    const prevState = get(tradeConfirmStateAtom)
+    const currentState = get(tradeConfirmStateAtom)
 
-    return { ...prevState, ...nextState }
+    if (nextState.isOpen === true) {
+      return {
+        isOpen: true,
+        error: null,
+        pendingTrade: null,
+        transactionHash: null, //
+      }
+    }
+
+    if (nextState.isOpen === false) {
+      return {
+        ...currentState,
+        isOpen: false,
+      }
+    }
+
+    if (nextState.error) {
+      return {
+        isOpen: currentState.isOpen,
+        error: nextState.error,
+        pendingTrade: null,
+        transactionHash: null, //
+      }
+    }
+
+    if (nextState.pendingTrade) {
+      return {
+        isOpen: currentState.isOpen,
+        error: null,
+        pendingTrade: nextState.pendingTrade,
+        transactionHash: null, //
+      }
+    }
+
+    if (nextState.transactionHash) {
+      return {
+        isOpen: currentState.isOpen,
+        error: null,
+        pendingTrade: null,
+        transactionHash: nextState.transactionHash,
+      }
+    }
+
+    return get(tradeConfirmStateAtom)
```

Comment:
Do we never update more than 1 property at a time?
Maybe it would be less error prone to have one atom per property updated, to be explicit?
Otherwise I feel like we could very well pass multiple properties expecting them all to be updated at once

Reply:
Makes sense! Thank you!
Fixed in the next PR: https://github.com/cowprotocol/cowswap/pull/2574/commits/22d65e0fa6f27b3399ecb2b61544c388b4438072